### ISSUE TITLE
Upgrade to 1.7.0 fails

### ISSUE THREAD

**Issue by marchenm** [Azure/aad-pod-identity#890]
**Have you**

- [x] Read [Troubleshooting Guide](../../docs/readmes/README.troubleshooting.md)
- [x] Read [Known Issues](../../docs/readmes/README.known-issues.md)
- [x] Searched on [GitHub issues](https://github.com/Azure/aad-pod-identity/issues)

**Describe the bug**
We're running a k8s cluster on Azure VMs using flannel. 

When running `helm upgrade` to 1.7.0 the nmi pod fails to start. 

I tried setting `nmi.allowNetworkPluginKubenet` to `true` but the results were the same. 

**Steps To Reproduce**
Install `aad-pod-identity` on a cluster with flannel. 

**Expected behavior**
nmi pod starts

**AAD Pod Identity version**
1.7.0 

**Kubernetes version**
1.18

**Additional context**
log output: 
```
{"ts":1606952117797.482,"msg":"failed to check if CNI plugin is kubenet, error: read /etc/default/kubelet: is a directory\n","v":0}
goroutine 1 [running]:
k8s.io/klog/v2.stacks(0xc000078201, 0xc00027ea20, 0x5b, 0x0)
	/go/pkg/mod/k8s.io/klog/v2@v2.3.0/klog.go:996 +0xb9
k8s.io/klog/v2.(*loggingT).output(0x2188a40, 0xc000000003, 0x186d1a0, 0xc000078240, 0xc000128a80, 0x20ebf7e, 0x7, 0x4e, 0x0)
	/go/pkg/mod/k8s.io/klog/v2@v2.3.0/klog.go:945 +0x191
k8s.io/klog/v2.(*loggingT).printf(0x2188a40, 0x3, 0x186d1a0, 0xc000078240, 0x16cce4a, 0x34, 0xc00063fea0, 0x1, 0x1)
	/go/pkg/mod/k8s.io/klog/v2@v2.3.0/klog.go:733 +0x17a
k8s.io/klog/v2.Fatalf(...)
	/go/pkg/mod/k8s.io/klog/v2@v2.3.0/klog.go:1463
main.main()
	/go/src/github.com/Azure/aad-pod-identity/cmd/nmi/main.go:78 +0x29b

goroutine 5 [chan receive]:
k8s.io/klog/v2.(*loggingT).flushDaemon(0x2188a40)
	/go/pkg/mod/k8s.io/klog/v2@v2.3.0/klog.go:1131 +0x8b
created by k8s.io/klog/v2.init.0
	/go/pkg/mod/k8s.io/klog/v2@v2.3.0/klog.go:416 +0xd8

goroutine 6 [select]:
go.opencensus.io/stats/view.(*worker).start(0xc00007c0f0)
	/go/pkg/mod/go.opencensus.io@v0.22.2/stats/view/worker.go:154 +0x105
created by go.opencensus.io/stats/view.init.0
	/go/pkg/mod/go.opencensus.io@v0.22.2/stats/view/worker.go:32 +0x57
```

helm values:
```yaml
installMICException: "true"

adminsecret: # removed values
  cloud: AzurePublicCloud
  subscriptionID: 
  resourceGroup: 
  vmType: standard
  tenantID: 
  clientID: msi
  clientSecret: msi 
  useMSI: "true"
  userAssignedMSIClientID: 

operationMode: "standard"

mic:
  logVerbosity: 6
  loggingFormat: json
  leaderElection:
    namespace: "kube-system"

nmi:
  loggingFormat: json
  micNamespace: "kube-system"
  metadataHeaderRequired: "true"

rbac:
  enabled: true
  allowAccessToSecrets: false
```

**Comment by chewong** [comment]
Looks like the kubelet config file `/etc/default/kubelet` is missing. I remember from https://github.com/Azure/aad-pod-identity/issues/743 that your cluster is not an AKS cluster. @aramase is there a way to determine whether the cluster is using kubenet without checking `/etc/default/kubelet`?

**Comment by chewong** [comment]
https://github.com/Azure/aad-pod-identity/pull/888 might be able to solve it

**Comment by aramase** [comment]
@chewong The kubelet configuration is the only source of truth on which CNI is configured. #888 will definitely help with this.

**Comment by marchenm** [comment]
I implemented the changes in #888 and that resolved the issue. Thanks for the quick response.
